config/trace.go

```diff
@@ -4,15 +4,174 @@
 package config // import "go.opentelemetry.io/contrib/config"
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
 	"go.opentelemetry.io/otel/trace/noop"
 )
 
-func initTracerProvider(cfg configOptions) (trace.TracerProvider, shutdownFunc) {
+var errNoValidSpanExporter = errors.New("no valid span exporter")
+
+func initTracerProvider(cfg configOptions, res *resource.Resource) (trace.TracerProvider, shutdownFunc, error) {
 	if cfg.opentelemetryConfig.TracerProvider == nil {
-		return noop.NewTracerProvider(), noopShutdown
+		return noop.NewTracerProvider(), noopShutdown, nil
+	}
+	opts := []sdktrace.TracerProviderOption{
+		sdktrace.WithResource(res),
+	}
+	for _, processor := range cfg.opentelemetryConfig.TracerProvider.Processors {
+		sp, err := spanProcessor(cfg.ctx, processor)
+		if err != nil {
+			return noop.NewTracerProvider(), noopShutdown, err
+		}
+		opts = append(opts, sdktrace.WithSpanProcessor(sp))
+	}
+	tp := sdktrace.NewTracerProvider(opts...)
+	return tp, tp.Shutdown, nil
+}
+
+func spanExporter(ctx context.Context, exporter SpanExporter) (sdktrace.SpanExporter, error) {
+	if exporter.Console != nil {
+		return stdouttrace.New(
```

Comment:
json schema validation for configuration that sets multiple exporters would fail, but it's true that it would be possible for someone to set multiple values through the model. i could add a check here for multiple exporters configured? WDYT?

Reply:
I think this would be a good check to add.

Reply:
added the check and a test for it